src/plugins/template/src/template_compiled_model.cpp

```diff
@@ -0,0 +1,247 @@
+        // Create input info
+        auto param_name = input.get_node()->get_friendly_name();
+        auto dims = get_dims(input);
+        InferenceEngine::TensorDesc desc(InferenceEngine::details::convertPrecision(input.get_element_type()),
+                                         dims,
+                                         InferenceEngine::TensorDesc::getLayoutByDims(dims));
+        auto data = std::make_shared<InferenceEngine::Data>(param_name, desc);
+        input_info = std::make_shared<InferenceEngine::InputInfo>();
+        input_info->setInputData(data);
+    }
+    auto& rt_info = input.get_rt_info();
+    auto it = rt_info.find("ie_legacy_preproc");
+    if (it != rt_info.end()) {
+        input_info->getPreProcess() = it->second.as<InferenceEngine::PreProcessInfo>();
+    }
+    it = rt_info.find("ie_legacy_td");
+    if (it != rt_info.end()) {
+        auto td = it->second.as<InferenceEngine::TensorDesc>();
+        input_info->getInputData()->reshape(td.getDims(), td.getLayout());
+        input_info->setPrecision(td.getPrecision());
+    }
+}
+void fill_output_info(const ov::Output<ov::Node>& output, InferenceEngine::DataPtr& output_info) {
+    if (!output_info) {
+        // Create input info
+        const auto& res_name = ov::op::util::create_ie_output_name(output);
+        auto dims = get_dims(output);
+        InferenceEngine::TensorDesc desc(InferenceEngine::details::convertPrecision(output.get_element_type()),
+                                         dims,
+                                         InferenceEngine::TensorDesc::getLayoutByDims(dims));
+        output_info = std::make_shared<InferenceEngine::Data>(res_name, desc);
+    }
+    auto& rt_info = output.get_rt_info();
+    auto it = rt_info.find("ie_legacy_td");
+    if (it != rt_info.end()) {
+        auto td = it->second.as<InferenceEngine::TensorDesc>();
+        output_info->reshape(td.getDims(), td.getLayout());
+        output_info->setPrecision(td.getPrecision());
+    }
+}
+}  // namespace legacy_convert
+}  // namespace ov
+
+// ! [executable_network:ctor_cnnnetwork]
+TemplatePlugin::CompiledModel::CompiledModel(const std::shared_ptr<const ov::Model>& model,
+                                             const std::shared_ptr<const ov::IPlugin>& plugin,
+                                             const InferenceEngine::ITaskExecutor::Ptr& task_executor,
+                                             const Configuration& cfg)
+    : ov::ICompiledModel(model, plugin, task_executor),  // Disable default threads creation
+      m_model(model->clone()),
+      _cfg(cfg) {
+    // TODO: if your plugin supports device ID (more that single instance of device can be on host machine)
+    // you should select proper device based on KEY_DEVICE_ID or automatic behavior
+    // In this case, _waitExecutor should also be created per device.
+    try {
+        compile_model(m_model);
+    } catch (const InferenceEngine::Exception&) {
+        throw;
```

Comment:
should we remove old exceptions?

Reply:
No because transformations can throw it.